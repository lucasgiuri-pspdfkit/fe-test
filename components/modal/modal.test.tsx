import { render, screen, fireEvent } from "@testing-library/react";
import Modal from "./modal";

const event = {
  id: "1",
  img: {
    src: "https://via.placeholder.com/400",
    alt: "placeholder",
  },
  title: "Test title",
  location: "Test location",
  date: {
    start: 1620000000000,
    end: 1620000000000,
  },
  coverUrl: "https://via.placeholder.com/400",
};

describe("Modal", () => {
  it("should render modal", () => {
    const onClose = jest.fn();
    const view = render(<Modal event={event} onClose={onClose} />);
    expect(screen.getByText("Test title")).toBeInTheDocument();
    expect(screen.getByText("Test location")).toBeInTheDocument();
    expect(view).toMatchSnapshot();
  });

  it("should call onClose when clicking outside modal", () => {
    const onClose = jest.fn();
    render(<Modal event={event} onClose={onClose} />);
    fireEvent.click(screen.getByTestId("modal"));
    expect(onClose).toHaveBeenCalledTimes(1);
  });
});
