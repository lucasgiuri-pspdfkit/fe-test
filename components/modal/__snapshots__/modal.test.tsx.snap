// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render modal 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fixed inset-0 flex items-center justify-center z-50 bg-white bg-opacity-60"
        data-testid="modal"
      >
        <div
          class="bg-white w-full max-w-md p-6 rounded-lg shadow-lg flex flex-col"
        >
          <div
            class="flex justify-center pb-6"
          >
            <img
              alt="placeholder"
              class="rounded-3xl overflow-hidden"
              data-nimg="1"
              decoding="async"
              height="400"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F400&w=828&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F400&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F400&w=828&q=75 2x"
              style="color: transparent;"
              width="400"
            />
          </div>
          <div
            class="flex flex-col gap-y-3"
          >
            <h3
              class="text-2xl
        font-avenirHeavy font-black
        text-gray-500"
            >
              Test title
            </h3>
            <p
              class="whitespace-pre-line  text-gray-500"
            >
              Test location
            </p>
            <div
              class="flex gap-x-2"
            >
              <h4
                class="text-lg
      
      text-gray-500"
              >
                3/5/2021 2:00:00
              </h4>
              <h4
                class="text-lg
      
      text-gray-500"
              >
                -
              </h4>
              <h4
                class="text-lg
      
      text-gray-500"
              >
                3/5/2021 2:00:00
              </h4>
            </div>
          </div>
          <div
            class="flex justify-center pt-6 lg:hidden"
          >
            <button
              class="bg-blue-100 text-white font-avenirHeavy py-3 px-7 rounded-lg"
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fixed inset-0 flex items-center justify-center z-50 bg-white bg-opacity-60"
      data-testid="modal"
    >
      <div
        class="bg-white w-full max-w-md p-6 rounded-lg shadow-lg flex flex-col"
      >
        <div
          class="flex justify-center pb-6"
        >
          <img
            alt="placeholder"
            class="rounded-3xl overflow-hidden"
            data-nimg="1"
            decoding="async"
            height="400"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F400&w=828&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F400&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F400&w=828&q=75 2x"
            style="color: transparent;"
            width="400"
          />
        </div>
        <div
          class="flex flex-col gap-y-3"
        >
          <h3
            class="text-2xl
        font-avenirHeavy font-black
        text-gray-500"
          >
            Test title
          </h3>
          <p
            class="whitespace-pre-line  text-gray-500"
          >
            Test location
          </p>
          <div
            class="flex gap-x-2"
          >
            <h4
              class="text-lg
      
      text-gray-500"
            >
              3/5/2021 2:00:00
            </h4>
            <h4
              class="text-lg
      
      text-gray-500"
            >
              -
            </h4>
            <h4
              class="text-lg
      
      text-gray-500"
            >
              3/5/2021 2:00:00
            </h4>
          </div>
        </div>
        <div
          class="flex justify-center pt-6 lg:hidden"
        >
          <button
            class="bg-blue-100 text-white font-avenirHeavy py-3 px-7 rounded-lg"
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
